lib/libutee/include/pta_scmi_client.h

```diff
@@ -16,7 +16,7 @@ enum optee_smci_pta_cmd {
 	/*
 	 * PTA_SCMI_CMD_CAPABILITIES - Get channel capabilities
 	 *
-	 * [out]    value[0].a: Capability bit mask (enum pta_scmi_caps)
+	 * [out]    value[0].a: Capabilities bit mask (PTA_SCMI_CAPS_*)
 	 * [out]    value[0].b: Extended capabilities or 0
 	 */
 	PTA_SCMI_CMD_CAPABILITIES = 0,
```

Comment:
Defining these as enums instead of the normal defines is possible. What is the gain? All generic interfaces will be expecting `uint32_t` anyway.

Reply:
I think looks better, as we did in [pkcs11_ta.h](https://github.com/OP-TEE/optee_os/blob/3.12.0/ta/pkcs11/include/pkcs11_ta.h#L57).

Reply:
And I think it looks worse, so on that point we're even. :-)
It has a few drawbacks too.
1. All the IDs must be declared next to each other
2. Complex interfaces using flags etc for various TA functions cannot reasonably be declared together with the function
3. The type of the value is less distinct compared to what can be done with a define.

Point 3 is an area we haven't bothered much with in the past, but I believe we're going to make all define constants using `U()` and friends in a not too distant future.

I don't think it's a disaster, but I wouldn't like to see it as good example of how to do things.
Generally I prefer the style used in https://github.com/OP-TEE/optee_os/blob/master/lib/libutee/include/pta_system.h

Reply:
makes sense. I can go this way.